core/Context.php

```diff
@@ -13,6 +13,23 @@
  */
 class Context
 {
+    public static function executeWithQueryParameters(array $parametersRequest, callable $callback)
+    {
+        // Temporarily sets the Request array to this API call context
+        $saveGET = $_GET;
+        $saveQUERY_STRING = @$_SERVER['QUERY_STRING'];
+        foreach ($parametersRequest as $param => $value) {
+            $_GET[$param] = $value;
+        }
```

Comment:
wouldn't it also need to set $_POST or unset it temporarily etc?

Reply:
Possibly, this code was moved from API\Proxy.php, I didn't change it. We can make it override `$_POST` too, just didn't want to accidentally break something.

Reply:
sorry... only just noticing... I wonder if it breaks functionality if we also did `$_GET = array();$_POST=array()` before the foreach? but I think it actually could break functionality. not needed for now I guess as it probably has always been that way.

Reply:
actually pretty sure it would break behaviour... this is why we currently use `$default = []` and needs to be changed there I suppose.